### ISSUE TITLE
Spurious wear sometimes detected outside known wear period

### ISSUE THREAD

**Issue by kitbeyer** [nimbal/vertdetach#1]
This is often caused by jostling or other movements while device is in transport.

van Hees has specifically addressed this issue previously in the Supplementary information of the following paper:

van Hees, V. T., Gorzelniak, L., Dean León, E. C., Eder, M., Pias, M., Taherian, S., Ekelund, U., Renström, F., Franks, P. W., Horsch, A., & Brage, S. (2013). Separating Movement and Gravity Components in an Acceleration Signal and Implications for the Assessment of Human Daily Physical Activity. PLoS ONE, 8(4). https://doi.org/10.1371/journal.pone.0061691

We should implement this and then test and modify accordingly.

**Comment by VertAdam** [comment]
I've made added an argument called _border_criteria_ in the dev branch (commit "basic border criteria added") and everything seems to be working well.

I have some concerns about the limitations of the criteria however and think that a small amount of work will be able to resolve the issues. 

Basically, the border algorithm as stated in van Hees' paper only talks about the "combined duration of their bordering non-wear periods". What this misses is that it could have one real and relatively short non-wear period within the final 6 hours of a participants collection and as long as it is followed by a long non-wear time at the end of the collection those valid 6 hours will be converted to non-wear. I have a quickly put together example from a sample below.
![image](https://user-images.githubusercontent.com/47859551/188235106-e049ee9f-2b15-4f3b-bf3b-046f049d2dbc.png)

I don't believe this will be difficult to fix and I already have an idea on how to do it (simply make sure that both sides of the surrounding non-wear periods are long in duration when compared to the enclosed wear time). I just need to code it up and make sure I'm not missing something.

**Comment by VertAdam** [comment]
I've committed the new method I mentioned above. 

Instead of looking at the total duration for the surrounding non-wear period, it looks at the shortest bordering non-wear duration and uses the same logic as Vanhees but halves the threshold (since we are now only looking at one border as opposed to the combined sum). 

For example, using the image used in VH supplementary info,
![image](https://user-images.githubusercontent.com/47859551/188285122-bca67278-2d3f-43ff-b0c2-14902cf153fa.png)

The original algorithm requires that if C is between 0 and 3 hours in length then if C is less 30 % of the bordering non-wear B+D, we change C to non-wear.

In the updated version we would only look at bordering duration B since it is smaller than D. In this case if C is between 0 and 3 hours in length then if C is less than 60% of the duration of B, C is changed to non-wear.

![image](https://user-images.githubusercontent.com/47859551/188285428-972e4bee-2cf3-4bda-806e-967d413f74f1.png)

**Comment by kitbeyer** [comment]
What if we make the two criteria additive, they are both required, and we open up the new criteria more so that in essence the minimum non-wear surrounding the wear has to be at least a third of the wear for any wear length up to 6 hours. I think this essentially just makes the new criterion [C/min(B, D) < 3]. Thoughts?

**Comment by kitbeyer** [comment]
Actually, after checking the paper I think these criteria are backwards - van Hees had <30% for 3-6 hour periods and < 80% for 0-3 hours. It sounds as if they ran it three times to account for additional non-wear affecting the remaining wear periods on the next pass through. They also had additional criteria for the final 24 hours that seem useful (I would add the option to include these for the first and last 24 hours) and for the first and last wear period.

I think that your implementation with the criteria flipped (<1.6 for up to 3 hours and <0.6 for 3-6 hours) combined with running it through for three iterations would probably catch some of the edge cases I was initially concerned about and still fix the case you described.

**Comment by VertAdam** [comment]
Good catch. I have now updated and changed the values as well as added the option for iterating through the border criteria as many times as requested. 

In summary the criteria are

For wear periods 0-3 hours in length, the wear period must be less than 160% of the duration of its shortest bordering non-wear period.

For wear periods 3-6 hours in length, the wear period must be less than 60% of the duration of its shortest bordering non-wear period.

When you get a chance if you can take another look at pull request #4 that would be great.
